action.yml

```diff
@@ -0,0 +1,106 @@
+name: Terraform Cloud Workspace
+description: Manages Terraform Cloud workspaces
+inputs:
+  terraform-version:
+    description: Terraform version
+    default: "1.0.3"
+  terraform-token:
+    description: Terraform Cloud token
+    required: true
+  terraform-host:
+    description: Terraform Cloud host
+    default: app.terraform.io
+  terraform-organization:
+    description: Terraform Cloud organization
+    required: true
+  name:
+    description: Name of the workspace
+    default: "${{ github.event.repository.name }}"
+  runner-terraform-version:
+    description: Terraform version used to create the workspace
+    default: "1.0.3"
+  environments:
```

Comment:
Avoid the use of `environment` in Terraform tools

https://www.terraform.io/docs/language/state/workspaces.html

Only "prefix" is used in Terraform and not "suffix" AFAIK. You could have `name` and `workspaces` where the latter are the workspaces as in `terraform workspace` and then you join. Make the join character an input as well and default to `-`.